File: src/main/java/easy/MinimumDepthBinaryTree.java
package easy;

import helpers.TreeNode;

public class MinimumDepthBinaryTree {
  public int solve(TreeNode root) {
    if (root == null) return 0;
    if (root.left == null && root.right == null) return 1;
    if (root.left == null) return 1 + solve(root.right);
    if (root.right == null) return 1 + solve(root.left);

    return 1 + Math.min(solve(root.left), solve(root.right));
  }
}

File: src/easy/MinimumDepthBinaryTree.swift
import helpers

func minimumDepthBinaryTree(_ root: TreeNode?) -> Int {
  guard let root = root else { return 0 }
  if root.left == nil && root.right == nil { return 1 }
  if root.left == nil {
    return 1 + minimumDepthBinaryTree(root.right)
  }
  if root.right == nil {
    return 1 + minimumDepthBinaryTree(root.left)
  }
  return 1 + min(minimumDepthBinaryTree(root.left), minimumDepthBinaryTree(root.right))

}
